src/theory/bv/rewrites

```diff
@@ -591,39 +591,24 @@
   )
   (def
     (n (+ x1in (bvsize x2)))
-    (x0n (ite (= x1i 0) x1in (- x1in (int.log2 x1i))))
-    (y0n (ite (= y1i 0) y1in (- y1in (int.log2 y1i))))
+    ; length(n) = log2(n) + 1, so we subtract 1 to compensate
+    (x0n (ite (= x1i 0) x1in (- x1in 1 (int.log2 x1i))))
+    (y0n (ite (= y1i 0) y1in (- y1in 1 (int.log2 y1i))))
   )
-  (and (> n 64) (> (- (* 2 n) (+ x0n y0n)) low))
+  (and (> n 64) (<= (- (* 2 n) (+ x0n y0n)) low))
   (extract high low (bvmul
     (concat (bv x1i x1in) x2)
     (concat (bv y1i y1in) y2)))
   (bv 0 (+ 1 (- high low))))
 
 ; (a udiv 2^k) ==> 0_k a[n-1: k]
-(define-cond-rule bv-udiv-pow2-1p
+(define-cond-rule bv-udiv-pow2-not-one
   ((x ?BitVec) (v Int) (n Int))
   (def (power (int.log2 v)))
   (and (int.ispow2 v) (> v 1))
   (bvudiv x (bv v n))
   (concat (bv 0 power) (extract (- n 1) power x)))
-(define-cond-rule bv-udiv-pow2-1n
```

Comment:
Why are these redundant? The only other rule on bvudiv seems to be bv-udiv-pow2-1p?

Reply:
If you look at `UdivPow2`, it never applies for negative cases.

Reply:
I added an explanation in the PR